backend/app/users/models.py

```diff
@@ -57,11 +53,42 @@ class UserDB(Base):
     is_active: Mapped[bool] = mapped_column(Boolean, nullable=False, default=True)
     is_verified: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False)
 
+    user_workspaces: Mapped[list["UserWorkspaceDB"]] = relationship(
+        "UserWorkspaceDB",
+        back_populates="user",
+        cascade="all, delete-orphan",
+        passive_deletes=True,
+    )
+    workspaces: Mapped[list["WorkspaceDB"]] = relationship(
+        "WorkspaceDB", back_populates="users", secondary="user_workspace", viewonly=True
+    )
+
     def __repr__(self) -> str:
         """Pretty Print"""
         return f"<{self.username} mapped to #{self.user_id}>"
 
 
+@dataclass
+class UserDBWithWorkspace:
```

Comment:
How about using a pydantic class instead?

Reply:
Removed as we don't need. Now, user and workspace is decoupled.